src/Libplanet.Types/Consensus/ValidatorSet.cs

```diff
@@ -264,16 +263,84 @@ public Validator GetProposer(long height, int round)
         /// <remarks>
         /// If <see cref="Vote.ValidatorPower"/> is null, power check is ignored.</remarks>
         /// <param name="blockCommit">The <see cref="BlockCommit"/> to check.</param>
-        /// <returns><see langword="true"/> if the <see cref="BlockCommit.Votes"/> is
-        /// ordered, <see langword="false"/> otherwise.</returns>
-        public bool ValidateBlockCommitValidators(BlockCommit blockCommit)
+        /// <exception cref="InvalidBlockCommitException">Thrown when some votes in the
+        /// <paramref name="blockCommit"/> does not have non-null power.</exception>
+        /// <exception cref="InvalidBlockCommitException">Thrown when validators from
+        /// <paramref name="blockCommit"/> is different from validators of this.</exception>
+        public void ValidateBlockCommitValidators(BlockCommit blockCommit)
         {
-            return Validators.Select(validator => validator.PublicKey)
-                       .SequenceEqual(
-                           blockCommit.Votes.Select(vote => vote.ValidatorPublicKey).ToList()) &&
-                   blockCommit.Votes.All(
-                       v => v.ValidatorPower is null ||
-                            v.ValidatorPower == GetValidator(v.ValidatorPublicKey).Power);
+            ValidatorSet validatorSetFromCommit = new ValidatorSet(blockCommit.Votes.Select(
+                v => v.ValidatorPower is BigInteger power
+                    ? new Validator(v.ValidatorPublicKey, power)
+                    : throw new InvalidBlockCommitException(
+                        "All votes in the block commit after block protocol version 10 " +
+                        "must have power.")).ToList());
```

Comment:
Do we need to construct `validatorSetFromCommit` variable?

Reply:
> Do we need to construct `validatorSetFromCommit` variable?

I don't think this code is necessary, because there is power comparison between the vote and the validator below.
Therefore, this code should be deleted.